### ISSUE TITLE
I am having difficulty running a dynamic ACM simulation from FOQUS

### ISSUE THREAD

**Issue by fsoepyan** [CCSI-Toolset/FOQUS#1019]
I tried running a dynamic ACM (Aspen Custom Modeler) simulation from FOQUS, but during the run, ACM would initialize the model, and then the run stops.

Can you please let me know what I can do to resolve the issue?

**Comment by ksbeattie** [comment]
Hi @fsoepyan!  Can you provide some more details?  Like the version of FOQUS, which model (your own, one of our examples, etc.), any log output, etc.?

**Comment by fsoepyan** [comment]
Thank you @ksbeattie for the prompt response, and for your help, I appreciate it.

Answering your questions:
- I downloaded FOQUS on 3/23/22 by going to github.com/ccsi-toolset/foqus, clicking the green "Code" button, and selecting "Download ZIP".
- I also downloaded the SimSinter Patch Release candidate 2.1.0.1 (https://github.com/CCSI-Toolset/SimSinter/releases/) and the Turbine 2.3.0.1 Patch Release Candidate (https://github.com/CCSI-Toolset/turb_sci_gate/releases/).
- I am using ACM Version 12.1.
- When I perform the runs, FOQUS would consider the run to be successful, but the run in ACM is not complete.
- When I run the dynamic ACM model from FOQUS, the messages that I received in ACM include:

SPARSE: No norm reduction after 20 step reductions
SPARSE: Return to best point
SPARSE: Exceeded maximum number of iterations.
...
SPARSE: Return to best point
Run complete 

- If the run had been done properly, after the "SPARSE" messages, the model should be integrating and continuing the run.

Please let me know if you need additional details.

**Comment by ksbeattie** [comment]
@fsoepyan generally speaking it is better if you install a released version of FOQUS, though that may not be the issue here.

Is this a model of yours, or one of our examples that you're working with?

**Comment by Ryan-Hughes-8** [comment]
@fsoepyan the sparse ACM messages that you included are common when the integrator or nonlinear equation solver within ACM fails. Did the dynamic model solve before connecting to FOQUS?

**Comment by fsoepyan** [comment]
@ksbeattie and @Ryan-Hughes-8: Thank you for your help and responses, I appreciate it.

Answering your questions:
- The dynamic ACM model is for a project I am working on, and therefore, it is not a CCSI model.
- When I run the dynamic ACM model in ACM (not in FOQUS), I managed to run the model successfully.

Please let me know if you need additional details.

**Comment by Ryan-Hughes-8** [comment]
@fsoepyan Ok, great. So it seems like the model may be just initializing and then closing. Are you specifying the time you want the dynamic simulation to run for? This can be done in the TimeSeries option in the FOQUS node.

Also, the SimSinter technical manual (https://github.com/CCSI-Toolset/SimSinter/tree/master/docs) has some nice info for setting up dynamic ACM models.

**Comment by ksbeattie** [comment]
@fsoepyan, going assume this is ok now and close this issue.  Feel free to re-open or open another issue if that's not the case.

**Comment by fsoepyan** [comment]
@ksbeattie and @Ryan-Hughes-8: I apologize for the late response. I managed to find a workaround for this issue:
1) I wrote a macro in Microsoft Excel to run the dynamic ACM model from Excel.
2) I uploaded this Excel file into SimSinter, and specified the macro from Step 1 as the macro to run.
3) I uploaded the .json file I created in Step 2 to Turbine in FOQUS.
4) I ran the Excel file from FOQUS, and both the Excel file and the dynamic ACM model ran successfully.

Yes, the issue can be closed.
